framework/src/engine/legacy/storage.ts

```diff
@@ -25,47 +31,93 @@ export class Storage {
 		this._db = db;
 	}
 
-	public async getBlockByID(id: Buffer): Promise<Buffer> {
-		return this._db.get(Buffer.concat([DB_KEY_BLOCKS_ID, id]));
+	// `ID` is the hashed value (utils.hash)
+	public async getTransactionByID(ID: Buffer): Promise<Buffer> {
+		return this._db.get(buildTxIDDbKey(ID));
```

Comment:
why we have upper case `ID`? i think the property name is lowercase `id` don't confuse it with ID in the function name as it is following camelCase
```suggestion
	// `id` is the hashed value (utils.hash)
	public async getTransactionByID(id: Buffer): Promise<Buffer> {
		return this._db.get(buildTxIDDbKey(id));
```